atlasdb-cassandra/src/test/java/com/palantir/atlasdb/cassandra/CassandraReloadableKeyValueServiceRuntimeConfigTest.java

```diff
@@ -0,0 +1,226 @@
+                .isInstanceOf(SafeIllegalArgumentException.class)
+                .hasLogMessage("'servers' must have at least one defined host");
+    }
+
+    @Test
+    public void mergedConfigPrioritisesInstallForConcurrentGetRangesThreadPoolSize() {
+        CassandraKeyValueServiceConfig config = configBuilderWithDefaultCredentialsAndReplicationFactor()
+                .concurrentGetRangesThreadPoolSize(1)
+                .build();
+
+        Refreshable<CassandraReloadableKeyValueServiceRuntimeConfig> reloadableConfig =
+                CassandraReloadableKeyValueServiceRuntimeConfig.fromConfigs(
+                        config, Refreshable.only(RUNTIME_CONFIG_WITH_DEFAULT_SERVERS));
+
+        assertThat(reloadableConfig.get().concurrentGetRangesThreadPoolSize()).isEqualTo(1);
+    }
+
+    @Test
+    public void mergedConfigDerivesThreadPoolSizeIfInstallIsEmpty() {
+        int poolSize = 10;
+        CassandraKeyValueServiceConfig config = configBuilderWithDefaultCredentialsAndReplicationFactor()
+                .poolSize(poolSize)
+                .build();
+        CassandraKeyValueServiceRuntimeConfig runtimeConfig =
+                runtimeConfigBuilder().servers(SERVERS_1).build();
+
+        Refreshable<CassandraReloadableKeyValueServiceRuntimeConfig> reloadableConfig =
+                CassandraReloadableKeyValueServiceRuntimeConfig.fromConfigs(config, Refreshable.only(runtimeConfig));
+
+        assertThat(reloadableConfig.get().concurrentGetRangesThreadPoolSize())
+                .isEqualTo(poolSize * SERVERS_1.numberOfThriftHosts());
+    }
+
+    @Test
+    public void mergedConfigPrioritisesInstallForDefaultGetRangesConcurrency() {
+        CassandraKeyValueServiceConfig config = configBuilderWithDefaultCredentialsAndReplicationFactor()
+                .concurrentGetRangesThreadPoolSize(10)
+                .build();
+
+        Refreshable<CassandraReloadableKeyValueServiceRuntimeConfig> reloadableConfig =
+                CassandraReloadableKeyValueServiceRuntimeConfig.fromConfigs(
+                        config, Refreshable.only(RUNTIME_CONFIG_WITH_DEFAULT_SERVERS));
+
+        assertThat(reloadableConfig.get().defaultGetRangesConcurrency())
+                .isEqualTo(config.defaultGetRangesConcurrency());
+    }
+
+    @Test
+    public void mergedConfigDerivesDefaultGetRangesConcurrencyIfInstallIsEmpty() {
+        CassandraKeyValueServiceConfig config = configBuilderWithDefaultCredentialsAndReplicationFactor()
+                .poolSize(10)
+                .build();
+
+        Refreshable<CassandraReloadableKeyValueServiceRuntimeConfig> reloadableConfig =
+                CassandraReloadableKeyValueServiceRuntimeConfig.fromConfigs(
+                        config, Refreshable.only(RUNTIME_CONFIG_WITH_DEFAULT_SERVERS));
+
+        assertThat(reloadableConfig.get().defaultGetRangesConcurrency())
+                .isEqualTo(Math.min(8, reloadableConfig.get().concurrentGetRangesThreadPoolSize() / 2));
```

Comment:
... what value do we actually expect here? I think it's 5; is that right?

Reply:
Nope, 8 -  min(8, (2 servers * 10 pool size)/2) = 8.
Have changed the test to show what the actual expected value is, but do let me know if you think the derivation is clearer.